### PR TITLE
 Prevent data loss when attaching to container

### PR DESCRIPTION
**Pull request by shin-**
Based on #1770 

@cjh1 Do you mind testing this PR and see if it solves your issue as well?

### CODE REVIEW

File: docker/transport/unixconn.py [hunk 1]
```diff
@@ -18,21 +18,46 @@
...
     def connect(self):
         sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
         sock.settimeout(self.timeout)
         sock.connect(self.unix_socket)
         self.sock = sock
 
+    def putheader(self, header, *values):
+        super(UnixHTTPConnection, self).putheader(header, *values)
+        if header == 'Connection' and 'Upgrade' in values:
+            self.disable_buffering = True
```
Comment (cjh1) [inline docker/transport/unixconn.py]: This is a much nicer approach, thanks.

**Comment by cjh1** [comment]
@shin- I can confirm that this also resolves me issue.

**Comment by shin-** [comment]
Great, thank you for confirming!

**Comment by cjh1** [comment]
@shin- Any idea when the next release will be? Thanks.

**Comment by shin-** [comment]
Should come out in the next couple of weeks 👍

**Comment by cjh1** [comment]
Awesome, thanks for helping me get this in.

**Pull request merged**